internal/controller/account/servicebinding/servicebinding.go

```diff
@@ -124,49 +161,157 @@ func (e *external) Create(ctx context.Context, mg resource.Managed) (managed.Ext
+	if e.isRotationEnabled(cr) {
+		btpName = servicebindingclient.GenerateRandomName(cr.Spec.ForProvider.Name)
+	} else {
+		btpName = cr.Spec.ForProvider.Name
+	}
+	cr.Spec.BtpName = &btpName
+
+	// Update the spec with the generated btpName
+	if err := e.kube.Update(ctx, cr); err != nil {
+		return managed.ExternalCreation{}, errors.Wrap(err, errUpdateStatus)
+	}
+
+	_, _, creation, err := e.instanceManager.CreateInstance(ctx, cr, *cr.Spec.BtpName)
+	if err != nil {
 		return managed.ExternalCreation{}, errors.Wrap(err, errCreateBinding)
 	}
 
-	return managed.ExternalCreation{
-		ConnectionDetails: managed.ConnectionDetails{},
-	}, nil
+	if err := e.kube.Status().Update(ctx, cr); err != nil {
+		return managed.ExternalCreation{}, errors.Wrap(err, errUpdateStatus)
+	}
+
+	// Remove force rotation annotation after successful creation
+	if cr.ObjectMeta.Annotations != nil {
+		if _, ok := cr.ObjectMeta.Annotations[servicebindingclient.ForceRotationKey]; ok {
+			meta.RemoveAnnotations(cr, servicebindingclient.ForceRotationKey)
+		}
+	}
+
+	return creation, nil
 }
 
-func (c *external) Update(ctx context.Context, mg resource.Managed) (managed.ExternalUpdate, error) {
-	_, ok := mg.(*v1alpha1.ServiceBinding)
+func (e *external) Update(ctx context.Context, mg resource.Managed) (managed.ExternalUpdate, error) {
+	cr, ok := mg.(*v1alpha1.ServiceBinding)
 	if !ok {
 		return managed.ExternalUpdate{}, errors.New(errNotServiceBinding)
 	}
-	if err := c.tfClient.Update(ctx); err != nil {
-		return managed.ExternalUpdate{}, err
+
+	// Check if current binding is already retired - if so, skip update as service bindings are immutable
+	currentBindingRetired := false
+	for _, retiredKey := range cr.Status.AtProvider.RetiredKeys {
+		if retiredKey.ID == cr.Status.AtProvider.ID {
+			currentBindingRetired = true
+			break
+		}
 	}
-	return managed.ExternalUpdate{
-		ConnectionDetails: managed.ConnectionDetails{},
-	}, nil
+
+	// Only update if the current binding is not retired (service bindings are immutable in BTP)
+	var updateResult managed.ExternalUpdate
+	if !currentBindingRetired {
+		// Use btpName from spec, fall back to name for backward compatibility
+		var btpName string
+		if cr.Spec.BtpName != nil {
+			btpName = *cr.Spec.BtpName
+		} else {
+			btpName = cr.Spec.ForProvider.Name
+		}
+
+		update, err := e.instanceManager.UpdateInstance(ctx, cr, btpName, cr.Status.AtProvider.ID)
+		if err != nil {
+			return managed.ExternalUpdate{}, err
+		}
+		updateResult = update
+	}
+
+	// Clean up expired keys if there are any retired keys
+	if cr.Status.AtProvider.RetiredKeys != nil {
+		if newRetiredKeys, err := e.keyRotator.DeleteExpiredKeys(ctx, cr); err != nil {
+			return managed.ExternalUpdate{}, errors.Wrap(err, errDeleteExpiredKeys)
+		} else {
+			cr.Status.AtProvider.RetiredKeys = newRetiredKeys
+			if err := e.kube.Status().Update(ctx, cr); err != nil {
+				return managed.ExternalUpdate{}, errors.Wrap(err, errUpdateStatus)
+			}
+		}
+	}
```

Comment:
```suggestion
	if cr.Status.AtProvider.RetiredKeys != nil {
		newRetiredKeys, err := e.keyRotator.DeleteExpiredKeys(ctx, cr)
		if err != nil {
			return managed.ExternalUpdate{}, errors.Wrap(err, errDeleteExpiredKeys)
		}
		cr.Status.AtProvider.RetiredKeys = newRetiredKeys
		if err := e.kube.Status().Update(ctx, cr); err != nil {
			return managed.ExternalUpdate{}, errors.Wrap(err, errUpdateStatus)
		}
	}
```
more go idiomatic --> less if in if in if